src/niftyone/figures/generator.py

```diff
@@ -30,13 +32,15 @@ def decorator(cls: type[T]) -> type[T]:
     return decorator
 
 
-def create_generator(view: str, query: str) -> "ViewGenerator":
+def create_generator(
+    view: str, join_entities: list[str], queries: list[str]
+) -> "ViewGenerator":
     """Function to create generator."""
 
     def _parse_view(view: str) -> tuple[str, dict[str, Any]]:
         """Parse view for figure-specific kwargs."""
-        # Match [view][(key1=value1,key2=value2,...)]
-        match = re.match(r"(\w+)\(([^=,]+=[^=,]+(?:,[^=,]+=[^=,]+)*)\)", view)
+        # Match [view][(key1=value1,key2=value2,key3=entities_dict,...)]
```

Comment:
We discussed this a bit before, but why not let the config structure transparently match the argument structure so no parsing is needed?

Reply:
I think that is doable - wanted to leave it out of this PR so can think about how to best implement it. I'm picturing that it would then require a key for each figure so that the arguments are passed appropriately for each figure. Right now, the config can handle generating multiple figures with the given queries.

Reply:
This is now updated in 2465a48 - @clane9 take a look at the updated config. its just passing the arguments directly so we can get rid of this altogether. I haven't tested this extensively yet, but it's passing the tests.